### PR TITLE
tech-debt: update SpotFleetRequest acceptance tests to use data source instance type

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #14383

Notes: 
* requests with  `m1.small` instance types consistently are not fulfilled so a switch here to using the `micro`'s seems to work and conveniently can be referenced from the data source in tests configs

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
tech-debt: update SpotFleetRequest acceptance tests to use data source instance type
```

Output from acceptance testing:

```
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (249.48s)
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (273.53s)
```

### CODE REVIEW

File: aws/resource_aws_spot_fleet_request_test.go [hunk 1]
```diff
@@ -641,9 +641,10 @@ func TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz(t *testing.T) {
 					testAccCheckAWSSpotFleetRequestExists(resourceName, &sfr),
 					resource.TestCheckResourceAttr(resourceName, "spot_request_state", "active"),
 					resource.TestCheckResourceAttr(resourceName, "launch_specification.#", "2"),
-					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "launch_specification.*", map[string]string{
-						"instance_type":     "m1.small",
-						"availability_zone": "us-west-2a",
+					tfawsresource.TestMatchTypeSetElemNestedAttrs(resourceName, "launch_specification.*", map[string]*regexp.Regexp{
+						// match Regexp as instance_type is one of the t2.micro or t3.micro preferred types
+						"instance_type":     regexp.MustCompile(`t(2|3).micro`),
```
Comment (bflad) [inline aws/resource_aws_spot_fleet_request_test.go:646]: Drive-by comment: You may find `tfawsresource.TestCheckTypeSetElemAttrPair()` easier to use in these cases, especially as we'll want to remove the hardcoded availability zones too 😄 

```go
					tfawsresource.TestCheckTypeSetElemAttrPair(resourceName, "launch_specification.*", "data.aws_ec2_instance_type_offering.available", "instance_type"),
```

Comment (anGie44) [reply aws/resource_aws_spot_fleet_request_test.go:646]: i thought so too! but i gave that one a try and from the output it doesn't seem to handle blocks like `launch_specification` ?  
```	
* Check 4/5 error: no TypeSet element "launch_specification.*", with value "t3.micro" 
```
even though you can see that value in the state that gets printed out :/

Comment (anGie44) [reply aws/resource_aws_spot_fleet_request_test.go:646]: @bflad, https://github.com/terraform-providers/terraform-provider-aws/pull/14843 could be an approach to supporting test checks with nested attributes like we have here.

Comment (bflad) [reply aws/resource_aws_spot_fleet_request_test.go:646]: Oh yikes, yeah, `TestCheckTypeSetElemAttrPair()` is for attribute TypeSet and not block TypeSet. Sorry about that. I think this is okay for now and we can iterate on it later if necessary. 👍
